relay-event-normalization/src/event.rs

```diff
@@ -201,6 +212,18 @@ pub fn normalize_event(event: &mut Annotated<Event>, config: &NormalizationConfi
         let _ =
             trimming::TrimmingProcessor::new().process_event(event, meta, ProcessingState::root());
     }
+
+    if config.is_last_normalize && config.remove_other.unwrap_or(!is_renormalize) {
```

Comment:
I find the behaviour strange if I explicitly state `remove_other: Some(true)` and then it does not run because it depends on `is_last_normalize` (additionally this is not documented).

Possibly include the check only in the `unwrap_or`.

Reply:
Using a single option in [59c0c95](https://github.com/getsentry/relay/pull/3097/commits/59c0c950929b63e64998e8ee817a21e7812b8dba).